ibm/service/database/resource_ibm_database.go

```diff
@@ -1710,11 +1763,25 @@ func resourceIBMDatabaseInstanceRead(context context.Context, d *schema.Resource
 	}
 	d.Set("auto_scaling", flattenICDAutoScalingGroup(autoSclaingGroup))
 
-	whitelist, err := icdClient.Whitelists().GetWhitelist(icdId)
-	if err != nil {
-		return diag.FromErr(fmt.Errorf("[ERROR] Error getting database whitelist: %s", err))
+	if _, ok := d.GetOk("whitelist"); ok {
```

Comment:
In Resource Read also don't use d.Get operations because the import functionality doesn't work

Reply:
Thanks for explanation
getOK does work in resource read  but it fails during the import scenario. Let say you have created  a database outside Terraform and want to import and  manage in future using Terraform. We need to import the resource. (https://learn.hashicorp.com/tutorials/terraform/state-import?in=terraform/state&utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS)
For import we first create a empty resource block
resource "ibm_database" "mydb" {
}
Run terraform import ibm_database.mydb <id>
Import also calls the same resource read since the block is empty d.GetOk fails for both whitelist and allowlist

Reply:
@hkantare thank you very very much for the explanation :D I've gone ahead and update the  code so that if whitelist isn't there we always use allowlist. I follow the terraform docs here: https://www.terraform.io/plugin/sdkv2/best-practices/deprecations#renaming-an-optional-attribute

So if `d.getOK("whitelist")` fails we'll always use allowlist. lmk what you think :D